hypervisor/qemu/qemu_process.go

```diff
@@ -13,6 +14,78 @@ import (
 	"github.com/hyperhq/runv/hypervisor"
 )
 
+type QemuLogFile struct {
+	name       string
+	offset     int64
+	cancelChan chan bool
+}
+
+func (f *QemuLogFile) Read(p []byte) (n int, err error) {
+	reader, err := os.Open(f.name)
+	defer reader.Close()
+	if err != nil {
+		return 0, err
+	}
```

Comment:
the defer and the if should be swapped.

and the `reader` is pushed to the sacked when the defer is invoked, but the reader can be changed after timeout, so you should wrapper the `reader.Close()` into a closure to delay the push the `reader`
```go
defer func() {
   reader.Close()
} ()
```

or just put the `reader.Close()` before every return.